pkg/cloud/services/network/carriergateways.go

```diff
@@ -0,0 +1,145 @@
+/*
```

Comment:
FYI reviewers: The Carrier Gateway (CAGW) re-conciliator is very similar [InternetGateway](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/network/gateways.go) resource, adapting to CAGW's requirements.